### PR TITLE
`lk` -> `llk` in `*sytrd_sy2sb`

### LINKED ISSUE #810: Nonstandard forward reference to 'lk' is not allowed in the same specification part causes compilation errors
### Description

There's some non-standard non-portable Fortran code at line 55569 of stdlib_linalg_lapack_s.fypp.

```
55569            logical(lk) :: lquery, upper
55570            integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
55571                       tpos, wpos, s2pos, s1pos
```
Forward reference to 'lk' is not allowed in the same specification part.  Some compilers will allow the use of 'lk' at line 55569 to reference the outer 'lk' and then from line 55570 onward use the 'lk' that is defined on line 55570.

### Expected Behaviour

The code would compile.   We see an error with XLF, NAG, and LLVM flang.

### Version of stdlib

33559f551d1c7f22ea2511e4f7e779e2aed410c3

### Platform and Architecture

Linux

### Additional Information

In the following subroutines, I changed the local declaration and all uses of 'lk' to 'llk' and things compiled fine.

```
stdlib_linalg_lapack_c.fypp:36510:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
stdlib_linalg_lapack_d.fypp:57095:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
stdlib_linalg_lapack_q.fypp:72920:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
stdlib_linalg_lapack_s.fypp:55570:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
stdlib_linalg_lapack_w.fypp:28871:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
stdlib_linalg_lapack_z.fypp:36925:           integer(ilp) :: i, j, iinfo, lwmin, pn, pk, lk, ldt, ldw, lds2, lds1, ls2, ls1, lw, lt,&
```

### PR DESCRIPTION
**Pull request by perazz**
Fix #810 @jvdp1 @sscalpone

Renaming this variable rather than the logical kind would minimize the number of changes to the code.

### CODE REVIEW

File: src/.stdlib_linalg_lapack_s.fypp.swp
Comment (jvdp1) [inline src/.stdlib_linalg_lapack_s.fypp.swp:1]: Could you remove this file, please?

**Review by jvdp1 — APPROVED** [review]
Thank you @perazz . Could you remof the hidden file, please?
Should we create a new release `v0.6.1` after merging this PR?

**Comment by perazz** [comment]
> Should we create a new release

I wouldn't think this is a critical update as it is more a standard compliance issue rather than a bug.

**Comment by sscalpone** [comment]
Thanks for the change.  It compiles fine in my sandbox.

**Comment by jvdp1** [comment]
Thank you @perazz . I will generate a new version `v0.6.1`, just to avoid that users try to compile the latest version (`v0.6.0`) and report the same error multiple times.

**Pull request merged**